### PR TITLE
Render shopping list items

### LINKED ISSUE #2: 2. As a user, I want to read all of my shopping lists and the items in them.
## Summary

The most important feature of our app is the ability to look at their shopping lists to see what's on them, so the user knows what they need to buy. For this to work, we'll have to be able to choose a our shopping lists. The `List` (the active shopping list) and `Home` (where the user will be able to switch shopping lists) components are the best place in our code to do that. In fact, this feature is already partially implemented. Refer to the acceptance criteria in this issue, and the existing code, to understand the work that remains.

The `List` component is rendered on the `/list` page. Because the navigation links are currently broken, you'll have to type that path into your browser's address bar.

## Acceptance criteria
- [x] The `useShoppingLists` hook is used in `App.jsx` to get the shopping lists a user has access to from the database
- [x] The lists are passed into the `Home` component as a prop named `data`
- [x] The `setListPath` function is passed into the `Home` component so we can update the current list
- [x] The `useShoppingListData` hook is used in `App.jsx` to get the items in the current list from the Firestore database
- [x] The list is passed into the `List` component as a prop named `data`
- [ ] In `Home.jsx`, the `SingleList` component is used to render the name of each shopping list a user is subscribed to
- [ ] In `List.jsx`, the `ListItem` component is used to render the name of each item

## Notes
- ⚠️ Your **only goal** is to make sure that the names of the items render as expected. You might be tempted to modify `ListItem.jsx` or `SingleList.jsx` to show more information – you will do that **later**.
- The variables you need are already being passed into the `List` and `Home` components. What steps do you need to take in order to render them?
- Additionally, until issues [4](https://github.com/the-collab-lab/tcl-78-smart-shopping-list/issues/4) and [5](https://github.com/the-collab-lab/tcl-78-smart-shopping-list/issues/5) you should only be able to see the shopping list that was manually added in [issue 1](https://github.com/the-collab-lab/tcl-78-smart-shopping-list/issues/1).

### PR DESCRIPTION
**Pull request by eva-lng**
_For an example of how to fill this template out, [see this Pull Request](https://github.com/the-collab-lab/tcl-3-smart-shopping-list/pull/44)._

## Description

SingleList and ListItem components imported into Home and List pages and implemented to render shopping lists and shopping list's items on those pages.

## Related Issue

closes #2 

## Acceptance Criteria

- [x] The `useShoppingLists` hook is used in `App.jsx` to get the shopping lists a user has access to from the database
- [x] The lists are passed into the `Home` component as a prop named `data`
- [x] The `setListPath` function is passed into the `Home` component so we can update the current list
- [x] The `useShoppingListData` hook is used in `App.jsx` to get the items in the current list from the Firestore database
- [x] The list is passed into the `List` component as a prop named `data`
- [x] In `Home.jsx`, the `SingleList` component is used to render the name of each shopping list a user is subscribed to
- [x] In `List.jsx`, the `ListItem` component is used to render the name of each item

## Type of Changes

`enhancement`

### Before

Screenshot 2024-08-15 at 20 51 53

Screenshot 2024-08-15 at 20 51 44

### After

Screenshot 2024-08-15 at 20 53 41

Screenshot 2024-08-15 at 20 53 49

## Testing Steps / QA Criteria

`npm start`
Navigate to the browser
Check Homepage to see the rendered shopping
Check the List page to see the rendered shopping list items names

### CODE REVIEW

**Review by didemydn — COMMENTED** [review]
Hi team :) Thank you 
the list page looks fine but I don't see the changes on the home page. 
please see the below 
image

image

File: src/views/Home.jsx [hunk 1]
```diff
@@ -7,10 +8,9 @@ export function Home({ data, setListPath }) {
 				Hello from the home (<code>/</code>) page!
 			</p>
 			<ul>
-				{/**
-				 * TODO: write some JavaScript that renders the `lists` array
-				 * so we can see which lists the user has access to.
-				 */}
+				{data.map((list, id) => (
+					<SingleList key={id} name={list.name} />
```
Comment (vivitt) [inline src/views/Home.jsx]: The `SingleList` component also receives path and `setListPath` as props. We should pass these props to ensure the list path is correctly stored and can be used to retrieve the list data.

Apart from that, this is very nice code. It's great that you added the `key` ✨

File: src/App.jsx [hunk 2]
```diff
@@ -38,7 +38,7 @@ export function App() {
 	 * This custom hook takes our token and fetches the data for our list.
 	 * Check ./api/firestore.js for its implementation.
 	 */
-	const data = useShoppingListData(listPath);
+	const data = useShoppingListData('/A45I0SLfsWeHLDg47ETZM5vP8fG2/test-list');
```
Comment (vivitt) [inline src/App.jsx]: Once `SingleList` receives all the props, we should be able to remove the hardcoded list path and replace it with `listPath` again.

**Review by vivitt — APPROVED** [review]
Great job here! Happy to see the app starting to grow 🚀

I left two comments for further improvement. Feel free to address them before merging or reach out if you have any questions.

**Pull request merged**